src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java

```diff
@@ -14,42 +14,49 @@
 class OAuthAuthorizationGrant extends AbstractMsalAuthorizationGrant {
 
     private AuthorizationGrant grant;
-    private final Map<String, List<String>> params;
-
-    /**
-     * init standard scopes
-     */
-    private OAuthAuthorizationGrant() {
-        params = new LinkedHashMap<>();
-
-        params.put(SCOPE_PARAM_NAME, Collections.singletonList(COMMON_SCOPES_PARAM));
-    }
+    private final Map<String, List<String>> params = new LinkedHashMap<>();
 
     OAuthAuthorizationGrant(final AuthorizationGrant grant, Set<String> scopesSet, ClaimsRequest claims) {
         this(grant, scopesSet != null ? String.join(" ", scopesSet) : null, claims);
     }
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant, String scopes, ClaimsRequest claims) {
-        this();
-        this.grant = grant;
-
+    String addCommonScopes(String scopes){
+        String allScopes = COMMON_SCOPES_PARAM;
 
         if (!StringHelper.isBlank(scopes)) {
-            this.scopes = scopes;
-            params.put(SCOPE_PARAM_NAME,
-                    Collections.singletonList(String.join(" ",params.get(SCOPE_PARAM_NAME)) + SCOPES_DELIMITER + scopes));
+            allScopes = allScopes + SCOPES_DELIMITER + scopes;
```

Comment:
Should we be merging these scopes instead? What if `scopes` contains some of the default scopes?

Reply:
STS will ignore duplicates but, probably we should do it as well